src/shared/components/query/quickSearch/QuickSearch.tsx

```diff
@@ -97,11 +102,21 @@ export default class QuickSearch extends React.Component {
         };
     }
 
+    @autobind
     private sampleToOption(sample: any, index: number): Option {
+        var studyName = 'Loading..';
+        if (this.props.queryStore.cancerStudies.isComplete) {
+            var study = this.props.queryStore.cancerStudies.result.find(
```

Comment:
once you pass in studies as a prop, you can make a computed which keys the studies by their id.  That will make lookup faster.   Here we are potentially iterating through every study in list for each sample.   It's probably negligible but since the list gets rendered on keystroke ...